sdk/azidentity/managed_identity_client.go

```diff
@@ -226,31 +226,31 @@ func (c *managedIdentityClient) createCloudShellAuthRequest(clientID string, sco
 }
 
 func (c *managedIdentityClient) getMSIType(ctx context.Context) (msiType, error) {
-	if c.msiType == unknown { // if we haven't already determined the msi type
+	if c.msiType == msiTypeUnknown { // if we haven't already determined the msi type
 		if endpointEnvVar := os.Getenv(msiEndpointEnvironemntVariable); endpointEnvVar != "" { // if the env var MSI_ENDPOINT is set
 			endpoint, err := url.Parse(endpointEnvVar)
 			if err != nil {
-				return unknown, err
+				return msiTypeUnknown, err
 			}
 			c.endpoint = endpoint
 			if secretEnvVar := os.Getenv(msiSecretEnvironemntVariable); secretEnvVar != "" { // if BOTH the env vars MSI_ENDPOINT and MSI_SECRET are set the MsiType is AppService
-				c.msiType = appService
+				c.msiType = msiTypeAppService
 			} else { // if ONLY the env var MSI_ENDPOINT is set the MsiType is CloudShell
-				c.msiType = cloudShell
+				c.msiType = msiTypeCloudShell
 			}
-		} else if c.imdsAvailable(ctx) { // if MSI_ENDPOINT is NOT set AND the IMDS endpoint is available the MsiType is Imds
+		} else if c.imdsAvailable(ctx) { // if MSI_ENDPOINT is NOT set AND the IMDS endpoint is available the MsiType is Imds. This will timeout after 500 milliseconds
```

Comment:
Weren't we going to change this to do it during MSI credential construction?  It can happen in a separate PR.

Reply:
Yup, I'm waiting for this PR to be completed. I have this change ready to go in a local branch.